lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -3392,6 +3392,158 @@ void mme_app_handle_handover_request_ack(
+          .bearer_contexts[i]
+          .eps_bearer_id = e_rab_admitted_list.item[i].e_rab_id;
+      s11_modify_bearer_request->bearer_contexts_to_be_modified
+          .bearer_contexts[i]
+          .s1_eNB_fteid.teid = e_rab_admitted_list.item[i].gtp_teid;
+      s11_modify_bearer_request->bearer_contexts_to_be_modified
+          .bearer_contexts[i]
+          .s1_eNB_fteid.interface_type = S1_U_ENODEB_GTP_U;
+      if (4 == blength(e_rab_admitted_list.item[i].transport_layer_address)) {
+        s11_modify_bearer_request->bearer_contexts_to_be_modified
+            .bearer_contexts[i]
+            .s1_eNB_fteid.ipv4 = 1;
+        memcpy(
+            &s11_modify_bearer_request->bearer_contexts_to_be_modified
+                 .bearer_contexts[i]
+                 .s1_eNB_fteid.ipv4_address,
+            e_rab_admitted_list.item[i].transport_layer_address->data,
+            blength(e_rab_admitted_list.item[i].transport_layer_address));
+      } else if (
+          16 == blength(e_rab_admitted_list.item[i].transport_layer_address)) {
+        s11_modify_bearer_request->bearer_contexts_to_be_modified
+            .bearer_contexts[i]
+            .s1_eNB_fteid.ipv6 = 1;
+        memcpy(
+            &s11_modify_bearer_request->bearer_contexts_to_be_modified
+                 .bearer_contexts[i]
+                 .s1_eNB_fteid.ipv6_address,
+            e_rab_admitted_list.item[i].transport_layer_address->data,
+            blength(e_rab_admitted_list.item[i].transport_layer_address));
+      } else {
+        OAILOG_ERROR_UE(
+            LOG_MME_APP, ue_context_p->emm_context._imsi64,
+            "Invalid IP address of %d bytes found for MME UE S1AP "
+            "Id: " MME_UE_S1AP_ID_FMT " (4 or 16 bytes was expected)\n",
+            blength(e_rab_admitted_list.item[i].transport_layer_address),
+            handover_notify_p->mme_ue_s1ap_id);
+        bdestroy_wrapper(&e_rab_admitted_list.item[i].transport_layer_address);
+        OAILOG_FUNC_OUT(LOG_MME_APP);
+      }
+      bdestroy_wrapper(&e_rab_admitted_list.item[i].transport_layer_address);
+      s11_modify_bearer_request->bearer_contexts_to_be_modified
+          .num_bearer_context++;
+
+      OAILOG_DEBUG_UE(
+          LOG_MME_APP, ue_context_p->emm_context._imsi64,
+          "Build MBR for ue_id %d\t bearer_id %d\t enb_teid %u\t sgw_teid %u\n",
+          ue_context_p->mme_ue_s1ap_id, bearer_id,
+          s11_modify_bearer_request->bearer_contexts_to_be_modified
+              .bearer_contexts[i]
+              .s1_eNB_fteid.teid,
+          current_bearer_p->s_gw_fteid_s1u.teid);
+    }
+
+    if (i == 0) {
+      cid = ue_context_p->bearer_contexts[EBI_TO_INDEX(bearer_id)]->pdn_cx_id;
+      pdn_context = ue_context_p->pdn_contexts[cid];
+      s11_modify_bearer_request->edns_peer_ip.addr_v4.sin_addr =
+          pdn_context->s_gw_address_s11_s4.address.ipv4_address;
+      s11_modify_bearer_request->teid = pdn_context->s_gw_teid_s11_s4;
+    }
+  }
```

Comment:
maybe add a comment after curly bracket to indicated end of for loop, e.g., `// end for loop for e_rab_admitted_list.no_of_items`.